src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -3328,14 +3292,85 @@ private IOperation CallNullableMember(IOperation value, SpecialMember nullableMe
         {
-            Debug.Assert(_currentConditionalAccessInstance != null);
-            IOperation result = _currentConditionalAccessInstance;
-            _currentConditionalAccessInstance = null;
-            return result;
+            Debug.Assert(!_currentConditionalAccessTracker.IsDefault && _currentConditionalAccessTracker.Operations.Count > 0);
+
+            IOperation testExpression = _currentConditionalAccessTracker.Operations.Pop();
+            return VisitConditionalAccessTestExpression(testExpression);
+        }
+
+        static bool IsConditionalAccessInstancePresentInChildren(IOperation operation)
+        {
+            if (operation is not NoneOperation)
+            {
+                return true;
+            }
+
+            // The conditional access should always be first leaf node in the subtree when performing a depth-first search. Visit the first child recursively
+            // until we either reach the bottom, or find the conditional access.
+            Operation currentOperation = (Operation)operation;
+            while (currentOperation.ChildOperations.GetEnumerator() is var enumerator && enumerator.MoveNext())
+            {
+                if (enumerator.Current is IConditionalAccessInstanceOperation)
+                {
+                    return true;
+                }
+
+                currentOperation = (Operation)enumerator.Current;
+            }
+
+            return false;
+        }
+
+        private IOperation VisitConditionalAccessTestExpression(IOperation testExpression)
+        {
+            Debug.Assert(!_currentConditionalAccessTracker.IsDefault);
+            SyntaxNode testExpressionSyntax = testExpression.Syntax;
+            ITypeSymbol? testExpressionType = testExpression.Type;
+
+            // In the case of repeated bad conditional accesses where the child node isn't present, we can end up
+            // in a situation in which there is no conditional access instance in the children of the test expression,
+            // but we are expecting the test expression to contain them. Because no nested IConditionalAccessInstance
+            // exists, we'll never hit the visit method to recurse into this again. To deal with that, we manually
+            // visit the child node in that case. This can happen for nested xml conditional accesses, see ConditionalAccessFlow_13
+            // in IOperationTests_IConditionalAccessExpression.vb. Any later flow access of the receiver is dropped,
+            // the known instance of this happening is tracked by https://github.com/dotnet/roslyn/issues/27564
+
+            if (_currentConditionalAccessTracker.Operations.Count > 0 && !IsConditionalAccessInstancePresentInChildren(testExpression))
+            {
+                _ = VisitConditionalAccessTestExpression(_currentConditionalAccessTracker.Operations.Pop());
```

Comment:
>_ = VisitConditionalAccessTestExpression(_currentConditionalAccessTracker.Operations.Pop()); [](start = 16, length = 92)

Is this code path reachable? It feels like similar check in ```VisitConditionalAccess``` should prevent us from getting here.

Reply:
I had thought so too, but I added it after `ConditionalAccessFlow_13` proved it was needed. Here's the original operation tree for the code (note the last conditional access with 2 none operation children):

```
IBlockOperation (3 statements) (OperationKind.Block, Type: null, IsInvalid) (Syntax: 'Sub M(c As  ... End Sub')
  IExpressionStatementOperation (OperationKind.ExpressionStatement, Type: null, IsInvalid) (Syntax: 'result = c?.X?.@a1?.<e1>')
    Expression: 
      ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: System.Object, IsInvalid, IsImplicit) (Syntax: 'result = c?.X?.@a1?.<e1>')
        Left: 
          IParameterReferenceOperation: result (OperationKind.ParameterReference, Type: System.Object) (Syntax: 'result')
        Right: 
          IConversionOperation (TryCast: False, Unchecked) (OperationKind.Conversion, Type: System.Object, IsInvalid, IsImplicit) (Syntax: 'c?.X?.@a1?.<e1>')
            Conversion: CommonConversion (Exists: False, IsIdentity: False, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
            Operand: 
              IConditionalAccessOperation (OperationKind.ConditionalAccess, Type: ?, IsInvalid) (Syntax: 'c?.X?.@a1?.<e1>')
                Operation: 
                  IParameterReferenceOperation: c (OperationKind.ParameterReference, Type: C) (Syntax: 'c')
                WhenNotNull: 
                  IConditionalAccessOperation (OperationKind.ConditionalAccess, Type: ?, IsInvalid) (Syntax: '.X?.@a1?.<e1>')
                    Operation: 
                      IFieldReferenceOperation: C.X As System.Xml.Linq.XElement (OperationKind.FieldReference, Type: System.Xml.Linq.XElement) (Syntax: '.X')
                        Instance Receiver: 
                          IConditionalAccessInstanceOperation (OperationKind.ConditionalAccessInstance, Type: C, IsImplicit) (Syntax: 'c')
                    WhenNotNull: 
                      IConditionalAccessOperation (OperationKind.ConditionalAccess, Type: ?, IsInvalid) (Syntax: '.@a1?.<e1>')
                        Operation: 
                          IOperation:  (OperationKind.None, Type: null) (Syntax: '.@a1')
                        WhenNotNull: 
                          IOperation:  (OperationKind.None, Type: null, IsInvalid) (Syntax: '.<e1>')
  ILabeledOperation (Label: exit) (OperationKind.Labeled, Type: null, IsImplicit) (Syntax: 'End Sub')
    Statement: 
      null
  IReturnOperation (OperationKind.Return, Type: null, IsImplicit) (Syntax: 'End Sub')
    ReturnedValue: 
      null
```

---
In reply to: [573257576](https://github.com/dotnet/roslyn/pull/51032#discussion_r573257576) [](ancestors = 573257576)

Reply:
It looks like there was a change in behavior reflected in ```ConditionalAccessFlow_13```, which was likely caused by the fix in ```IsConditionalAccessInstancePresentInChildren```. Do we still need this check here after the fix? Could you please elaborate why?

---
In reply to: [573954488](https://github.com/dotnet/roslyn/pull/51032#discussion_r573954488) [](ancestors = 573954488,573257576)